multicast/src/main/kotlin/com/dropbox/flow/multicast/Multicaster.kt

```diff
@@ -66,39 +64,25 @@ class Multicaster<T>(
 
     private val channelManager by lazy(LazyThreadSafetyMode.SYNCHRONIZED) {
         ChannelManager(
-                scope = scope,
-                bufferSize = bufferSize,
-                onActive = {
-                    SharedFlowProducer(
-                            scope = scope,
-                            src = source(),
-                            channelManager = it
-                    )
-                },
-                piggybackingDownstream = piggybackingDownstream,
-                onEach = onEach
+            scope = scope,
+            bufferSize = bufferSize,
+            upstream = source,
+            piggybackingDownstream = piggybackingDownstream,
+            onEach = onEach
```

Comment:
just a note for the future; i feel like making this onEach option, wdyt ? (not a change request for this CL :) )

Reply:
yeah, it could be. Though at the end of the day, multicaster is an internal class so no big deal either way